include/CubatureContext2D.hpp

```diff
@@ -13,65 +13,63 @@ namespace blitzdg {
 	private:
 		index_type NCubature_;
         index_type NumCubaturePoints_;
-        real_vec_smart_ptr r_;
-        real_vec_smart_ptr s_;
-        real_vec_smart_ptr w_;
-        real_mat_smart_ptr V_;
-        real_mat_smart_ptr rx_;
-        real_mat_smart_ptr sx_;
-        real_mat_smart_ptr ry_;
-        real_mat_smart_ptr sy_;
-        real_mat_smart_ptr J_;
-        real_mat_smart_ptr Dr_;
-        real_mat_smart_ptr Ds_;
-        std::unique_ptr<real_tensor3_type> MM_;
-        std::unique_ptr<real_tensor3_type> MMChol_;
-        real_mat_smart_ptr x_;
-        real_mat_smart_ptr y_;
-        real_mat_smart_ptr W_;
-
-
+        std::shared_ptr<real_vector_type> r_;
+        std::shared_ptr<real_vector_type> s_;
+        std::shared_ptr<real_vector_type> w_;
+        std::shared_ptr<real_matrix_type> V_;
+        std::shared_ptr<real_matrix_type> rx_;
+        std::shared_ptr<real_matrix_type> sx_;
+        std::shared_ptr<real_matrix_type> ry_;
+        std::shared_ptr<real_matrix_type> sy_;
+        std::shared_ptr<real_matrix_type> J_;
+        std::shared_ptr<real_matrix_type> Dr_;
+        std::shared_ptr<real_matrix_type> Ds_;
+        std::shared_ptr<real_tensor3_type> MM_;
+        std::shared_ptr<real_tensor3_type> MMChol_;
+        std::shared_ptr<real_matrix_type> x_;
+        std::shared_ptr<real_matrix_type> y_;
+        std::shared_ptr<real_matrix_type> W_;
 
     public:
         CubatureContext2D() = default;
         CubatureContext2D(
             index_type NCubature,
             index_type NumCubaturePoints,
```

Comment:
Just curious why you decided to change the constructor parameter types?

Reply:
I don't believe it's necessary... at one point I convinced myself it was necessary or a good idea, but I don't recall my reasoning now.